bytes/bytes/raw/file_raw_repository.py

```diff
@@ -62,3 +72,63 @@ def _raw_file_path(self, raw_id: UUID, boefje_meta: BoefjeMeta) -> Path:
 
     def _index(self, raw_id: UUID) -> str:
         return str(raw_id)[: self.UUID_INDEX]
+
+
+class S3RawRepository(RawRepository):
+    def __init__(
+        self,
+        file_middleware: FileMiddleware,
+        bucket_per_org: bool,
+        s3_bucket_prefix: str = "OpenKAT-",
+        s3_bucket_name: str = "OpenKAT",
+    ) -> None:
+        self._file_middleware = file_middleware
+        self.bucket_per_org = bucket_per_org
+        self.s3_bucket_prefix = s3_bucket_prefix
+        self.s3_bucket_name = s3_bucket_name
+
+    @cached_property
+    def _s3resource(self):
+        session = BotoSession()
+        return session.resource("s3")
+
+    def get_or_create_bucket(self, organization: str):
+        # Create a bucket, and if it exists already return that instead
+        bucket_name = self.s3_bucket_name
+        if self.bucket_per_org:
+            bucket_name = f"{self.s3_bucket_prefix}{organization}"
+            try:
+                bucket = self._s3resource.create_bucket(Bucket=bucket_name)
+                bucket.wait_until_exists()
+                return bucket
+            except Exception as error:
+                logger.error("Something went wrong with creating bucket %s: %s", bucket_name, error)
+                raise error
+        return self._s3resource.Bucket(name=bucket_name)
+
+    def save_raw(self, raw_id: UUID, raw: RawData) -> None:
+        file_name = self._raw_file_name(raw_id, raw.boefje_meta)
+        contents = self._file_middleware.encode(raw.value)
+
+        logger.info("Writing raw data with id %s to s3", raw_id)
+        bucket = self.get_or_create_bucket(raw.boefje_meta.organization)
+        bucket.Object(file_name).put(Body=contents)
+
+    def get_raw(self, raw_id: UUID, boefje_meta: BoefjeMeta) -> RawData:
+        file_name = self._raw_file_name(raw_id, boefje_meta)
+        bucket = self.get_or_create_bucket(boefje_meta.organization)
+
+        try:
+            contents = bucket.Object(file_name).get()["Body"].read()
+        except self._s3resource.meta.client.exceptions.NoSuchBucket as error:
```

Comment:
(Perhaps here we could catch whether the bucket for the organization wasn't created yet and create it in that case?)

Reply:
Usually, buckets cannot be created by the same user who accesses them.

Reply:
So a bucket should not be created by the OpenKAT instance ever?

Reply:
For now with [264733a](https://github.com/minvws/nl-kat-coordination/pull/3505/commits/264733a1d127df640eaabbcad9c9b1b76cfe5538) a bucket should always get created by `S3RawRepository.get_or_create_bucket()` and if it does not, it should throw an exception before this piece of code.

Reply:
> If we expect users to not use this functionality and create buckets outside of OpenKAT, we could perhaps drop bucket creation?

I think giving the user the responsibility to create buckets is more than fair. *Unless* they have chosen to have `bucket_per_org` enabled. Because in that case we might want to give that responsibility to OpenKAT to make adding new organisations less troublesome. Or do you think it should still be the job of the user then?

Reply:
Perhaps we can also make it configurable whether OpenKAT should create buckets? It makes the PR a bit more tricky in terms of configuration, so I'd say that we do need to know if creating buckets in OpenKAT is a requirement, which might not be the case (yet)? If not, we just assume the buckets are present and raise a descriptive error if they are not.